### ISSUE TITLE
Type for NextAuthOptions.adapter and PrismaAdapter gives error due to possiblity of `undefined`

### ISSUE THREAD

**Issue by devjiwonchoi** [nextauthjs/next-auth#8136]
### Question 💬

Error in Lint (modified to be readable):

```mdx
Type (@auth/core/adapters).Adapter is not assignable to type (next-auth/adapters").Adapter
Types of property 'createUser' are incompatible.

Type '((user: Omit<AdapterUser, "id">) => Awaitable<AdapterUser>) | undefined'
is not assignable to type '(user: Omit<AdapterUser, "id">) => Awaitable<AdapterUser>'.
      
Type 'undefined' is not assignable to type '(user: Omit<AdapterUser, "id">) => Awaitable<AdapterUser>'.ts(2322)

types.d.ts(106, 5): The expected type comes from property 'adapter' which is declared here on type 'AuthOptions'

(property) AuthOptions.adapter?: Adapter | undefined
```

### Code

```ts
import { NextAuthOptions } from 'next-auth'
import EmailProvider from 'next-auth/providers/email'
import { PrismaAdapter } from '@auth/prisma-adapter'
import { PrismaClient } from '@prisma/client'

const prisma = new PrismaClient()

export const authOptions: NextAuthOptions = {
  adapter: PrismaAdapter(prisma), // error here
  providers: [
    EmailProvider({
      server: {
        host: process.env.GMAIL_SMTP_HOST,
        port: process.env.GMAIL_SMTP_PORT,
        auth: {
          user: process.env.GMAIL_USER_EMAIL,
          pass: process.env.GMAIL_APP_PASSWORD,
        },
      },
      from: process.env.EMAIL_FROM,
    }),
  ],
}
```

### Additional Info

```console
$ npx prisma generate
Prisma schema loaded from schema.prisma

✔ Generated Prisma Client (5.0.0 | library) to ./node_modules/.pnpm/@prisma+client@5.0.0_prisma@5.0.0/node_modules/@prisma/client in 64ms
You can now start using Prisma Client in your code. Reference: https://pris.ly/d/client

import { PrismaClient } from '@prisma/client'
const prisma = new PrismaClient()
```

### Reference

- [NextAuth.js PrismaAdapter](https://authjs.dev/reference/adapter/prisma#prismaadapter)

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by balazsorban44** [comment]
This has been fixed in #8054, make sure you use the latest of `next-auth`

**Comment by mohammedkharoda** [comment]
`adapter: PrismaAdapter(prisma) as Adapter,` ' use this and it will work'
